zebra/zebra_nhg.c

```diff
@@ -122,6 +122,33 @@ static void nexthop_set_resolved(afi_t afi, const struct nexthop *newhop,
 	_nexthop_add(&nexthop->resolved, resolved_hop);
 }
 
+/* Checks if nexthop we are trying to resolve to is valid */
+static bool nexthop_valid_resolve(const struct nexthop *nexthop,
```

Comment:
nit: maybe the function's name can be more specific if this only valid for ifindex'ed nexthops?

Reply:
The first check is valid for all nexthop types.

But I named it that with the expectation that someone in the future may want to add checks of valid/invalid nexthops for other nexthop types as well.